<launch>

    <!-- color conversion from YUV422 to RGB -->
    <node pkg="ti_viz_nodes" type="viz_color_conv_yuv2rgb" name="viz_color_conv_yuv2rgb_node" output="screen" args="" required="true">
        <param name="width" value="1280"/>
        <param name="height" value="720"/>
        <param name= "input_yuv_topic"    value = "camera/right/image_raw"/>
        <param name= "output_rgb_topic"   value = "camera/right/image_raw_rgb"/>
        <param name= "yuv_format"   value = "YUV422"/>
    </node>

    <!-- generate disparity map and confidence map from disparity_raw  -->
    <include file="$(find ti_viz_nodes)/launch/viz_disparity.launch"/>

    <!-- color conversion for image_rect_nv12 for visualization -->
    <node pkg="ti_viz_nodes" type="viz_color_conv_yuv2rgb" name="viz_color_conv_yuv2rgb_node_rect" output="screen" args="" required="true">
        <param name="width"            value="1280"/>
        <param name="height"           value="720"/>
        <param name="input_yuv_topic"  value="camera/right/image_rect_nv12"/>
        <param name="output_rgb_topic" value="camera/right/image_rect_rgb"/>
        <param name="yuv_format"       value="YUV420"/>
        <param name="yuv420_luma_only" value="true"/>
    </node>

    <!-- viz_semseg  -->
    <node pkg="ti_viz_nodes" type="viz_semseg" name="viz_semseg" output="screen" args="" required="true">
        <param name="rectified_image_topic"    value="camera/right/image_rect_rgb"/>
        <param name="vision_cnn_tensor_topic"  value="vision_cnn/tensor"/>
        <param name="vision_cnn_image_topic"   value="vision_cnn/out_image"/>
    </node>

    <!-- rviz -->
    <node pkg="rviz" type="rviz" name="rviz_sde" args="-d $(find ti_viz_nodes)/rviz/sde_semseg.rviz" />

</launch>
